test/cppcheck_suppressions.txt

```diff
@@ -25,48 +25,11 @@
 shiftNegative:src/utils/msc_tree.cc
 *:src/utils/acmp.cc
 *:src/utils/msc_tree.cc
-invalidScanfArgType_int:src/rules_set_properties.cc:101
-invalidScanfArgType_int:src/rules_set_properties.cc:102
 
 
 // 
 // ModSecurity v3 code...
 // 
-unmatchedSuppression:src/utils/geo_lookup.cc:82
-useInitializationList:src/utils/shared_files.h:87
-unmatchedSuppression:src/utils/msc_tree.cc
-functionStatic:headers/modsecurity/transaction.h:408
-duplicateBranch:src/audit_log/audit_log.cc:226
-unreadVariable:src/request_body_processor/multipart.cc:435
-stlcstrParam:src/audit_log/writer/parallel.cc:145
```

Comment:
I didn't checked all lines here, so now just take a look only at this line. I can't find the reason why is this line here. You didn't add any in-place note  (`cppcheck-...`) over there, but the tests were success.

Does this mean that this file was totally unmaintained?

(Again: this is just an example.)

Reply:
In the case of audit_log.cc:226 & parallel.cc:145 it seems the suppression was no longer necessary.

That is, like in the other entries, I went to that line, reviewed why the suppression was necessary, and replaced it with an inline suppression. In some cases, it seemed the line was incorrect or the suppression no longer applied, so I removed the suppression without adding an inline suppression and checked whether cppcheck reported it again.

I think this goes back to one of the arguments I made about preferring inline suppressions, which is that if you change code that does away with the need of a current suppression, you won't get feedback about that, and thus the file becomes stale with no longer needed entries.